code/datums/components/caltrop.dm

```diff
@@ -1,16 +1,29 @@
 /datum/component/caltrop
+	///Minimum damage when crossed
 	var/min_damage
+
+	///Maximum damage when crossed
 	var/max_damage
+
+	///Probability of stunning and doing daamge
 	var/probability
+
+	///Duration of weaken when crossed
+	var/weaken_duration
+
+	///Shoebypassing, walking interaction, silence
 	var/flags
 
+	///TODO: soundfile?
+
 	var/cooldown = 0
 
-/datum/component/caltrop/Initialize(_min_damage = 0, _max_damage = 0, _probability = 100,  _flags = NONE)
-	min_damage = _min_damage
-	max_damage = max(_min_damage, _max_damage)
-	probability = _probability
-	flags = _flags
+/datum/component/caltrop/Initialize(min_damage = 0, max_damage = 0, probability = 100, weaken_duration = 6 SECONDS, flags = NONE)
+	src.min_damage = min_damage
+	src.max_damage = max(min_damage, max_damage)
+	src.probability = probability
+	src.weaken_duration = weaken_duration
+	src.flags = flags
```

Comment:
Could maybe revert the changes to the argument removal of the `_`? Imo the previous version is more readable than this